spyder/plugins/base.py

```diff
@@ -136,6 +136,10 @@ def closeEvent(self, event):
         self.plugin.dockwidget.setVisible(True)
         self.plugin.switch_to_plugin()
         QMainWindow.closeEvent(self, event)
+        # Qt might want to do something with this soon,
+        # So it should not be deleted by python yet.
+        # Fixes spyder-ide/spyder#10704
+        self.plugin.__unsafe__window = self
```

Comment:
Does this affect memory footprint of spyder. I mean by keeping a reference to that windows (it should be small anyway) but just checking.

Reply:
It is not an ideal solution as it indeed affect the footprint of spyder. However the only thing kept here is the python wrapper as the C++ object is destroyed. I would therefore expect the footprint to be negligible.

Reply:
👍 in other Qt apps I have developed I kept a timer to periodically delete these references, we could eventually have something like that. 

Anyway this seems like a good fix in the meantime!